rest_framework/exceptions.py

```diff
@@ -65,15 +73,61 @@ def __str__(self):
 # from rest_framework import serializers
 # raise serializers.ValidationError('Value was invalid')
 
+ErrorDetails = namedtuple('ErrorDetails', ['message', 'code'])
+
+
 class ValidationError(APIException):
     status_code = status.HTTP_400_BAD_REQUEST
+    code = None
 
-    def __init__(self, detail):
-        # For validation errors the 'detail' key is always required.
-        # The details should always be coerced to a list if not already.
-        if not isinstance(detail, dict) and not isinstance(detail, list):
-            detail = [detail]
-        self.detail = _force_text_recursive(detail)
+    def __init__(self, detail, code=None):
+        if code:
+            self.full_details = ErrorDetails(detail, code)
+        else:
+            self.full_details = detail
```

Comment:
The population of `.full_details`. Should just be storing the message and code at this point.
(Let me know if I'being braindead - I only get a brief amount of time on each review)